Net core api

zkonstr opened the pull request.

zkonstr committed:
gitignore added

zkonstr committed:
feat: ASP.NET Core api created

zkonstr committed:
feat: User controller and DB implemented

Added Create, Updete, Delete, Get by id/ Get all operations for user

mrZmitser commented on DebtNote/Controllers/UserController.cs:
```diff
@@ -0,0 +1,57 @@
+        private readonly ApplicationContext _context;
+
+        public UserController(ApplicationContext context) => _context = context;
+        [HttpGet]
```
Add spaces between the methods here and further

mrZmitser commented on DebtNote/Database/Migrations/20220617120138_createdatabase.cs:
```diff
@@ -0,0 +1,48 @@
+﻿using Microsoft.EntityFrameworkCore.Migrations;
+
+#nullable disable
```
explain this

zkonstr replied:
I don't know anything about migrations, but it helps me to create and fill tables in DB by one command

mrZmitser replied:
it's not about migrations
https://docs.microsoft.com/en-us/dotnet/csharp/nullable-references

mrZmitser commented on DebtNote/Database/Migrations/20220617120138_createdatabase.cs:
```diff
@@ -0,0 +1,48 @@
+
+namespace DebtNote.Database.Migrations
+{
+    public partial class createdatabase : Migration
```
what about classes name convention? rename to CreatedDatabase

zkonstr replied:
renamed

mrZmitser replied:
commit! :)

mrZmitser commented on DebtNote/Database/Migrations/20220617120138_createdatabase.cs:
```diff
@@ -0,0 +1,48 @@
+                constraints: table =>
+                {
+                    table.PrimaryKey("PK_Payments", x => x.Id);
+                });
```
move the closing bracket `)` to the next line

zkonstr replied:
this file was auto-generated, but... Well, I'll move it

mrZmitser replied:
well, sorry, if it's generated do nothing here, leave as it was

mrZmitser commented on DebtNote/Database/Migrations/20220617120138_createdatabase.cs:
```diff
@@ -0,0 +1,48 @@
+                constraints: table =>
+                {
+                    table.PrimaryKey("PK_Users", x => x.Id);
+                });
```
and here

mrZmitser commented on DebtNote/Database/Migrations/20220617120138_createdatabase.cs:
```diff
@@ -0,0 +1,48 @@
+        protected override void Down(MigrationBuilder migrationBuilder)
+        {
+            migrationBuilder.DropTable(
+                name: "Payments");
```
and here and further

mrZmitser commented on DebtNote/Models/User.cs:
```diff
@@ -0,0 +1,14 @@
+        [Required]
+        public int Id { get; set; }
+        [Required]
+        public string? Name { get; set; }
```
Why is name nullable?

zkonstr replied:
To much warnings about it.

mrZmitser commented on DebtNote/Services/Implimentations/TransferService.cs:
```diff
@@ -0,0 +1,50 @@
+            return Users;
+        }
+
+        public void Transfer(int amount, IUserRepository<User>? users)
```
What the hell is going here? Why is it not in User Service? Why it receives a users repo?

zkonstr replied:
It was just yesterday first try to use services. Now it's deleted

mrZmitser commented on DebtNote/Repositories/Implimentations/PaymentRepository.cs:
```diff
@@ -0,0 +1,33 @@
+
+namespace DebtNote.Repositories.Implimentations
+{
+    public class PaymentRepository<Payment> : IPaymentRepository<Payment>
```
What the sense of generics here?

mrZmitser commented on DebtNote/Repositories/Implimentations/UserRepository.cs:
```diff
@@ -0,0 +1,39 @@
+
+namespace DebtNote.Repositories.Implimentations
+{
+    public class UserRepository<User> : IUserRepository<User>
```
What the sense of generics here?

mrZmitser commented on DebtNote/Repositories/Implimentations/UserRepository.cs:
```diff
@@ -0,0 +1,39 @@
+﻿using DebtNote.Database;
+using DebtNote.Repositories.Interfaces;
+
+namespace DebtNote.Repositories.Implimentations
```
implEmentation

mrZmitser commented on DebtNote/Services/Implimentations/TransferService.cs:
```diff
@@ -0,0 +1,50 @@
+        public void Transfer(int amount, IUserRepository<User>? users)
+        {
+
+            User user1 = users.Create(new User
```
why do you create users here, not in a higher level?

zkonstr replied:
just for test

mrZmitser reviewed (changes requested):
Reply to all the comments or fix stuff where required

zkonstr committed:
feat: user service implemented

some functions from UserController were moved to the UserService
Repositories have been deleted
